lib/_http_incoming.js

```diff
@@ -47,7 +47,7 @@ function IncomingMessage(socket) {
   // response (client) only
   this.statusCode = null;
   this.statusMessage = null;
-  this._client = socket; // deprecated
+  this.client = socket;
```

Comment:
I'd say that `socket` and `connection` are used far more than `client`, so I would hold off on making deprecation messages for them.

Regarding `Object.defineProperty()` in the constructor, did you try this change with the `request` module to make sure it is ok?